src/plugins/orangehrmBuzzPlugin/Api/BuzzLikeOnShareAPI.php

```diff
@@ -0,0 +1,360 @@
+     *     ),
+     *     @OA\Response(
+     *         response="200",
+     *         description="Success",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="data",
+     *                 ref="#/components/schemas/Buzz-BuzzLikeOnShareModel"
+     *             ),
+     *             @OA\Property(
+     *                 property="meta",
+     *                 type="object",
+     *                 @OA\Property(property="total", type="integer")
+     *             )
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response="422",
+     *         description="Unprocessable Content - Invalid Share ID",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="error",
+     *                 type="object",
+     *                 @OA\Property(property="status", type="string", default="422"),
+     *                 @OA\Property(property="message", type="string", default="Invalid Parameter"),
+     *                 @OA\Property(
+     *                     property="data",
+     *                     type="object",
+     *                     @OA\Property(
+     *                         property="invalidParamKeys",
+     *                         type="array",
+     *                         @OA\Items(default="shareId")
+     *                     )
+     *                 )
+     *             )
+     *         )
+     *     ),
+     *     @OA\Response(
+     *         response="400",
+     *         description="Bad Request - Liking a post that is already liked",
+     *         @OA\JsonContent(
+     *             @OA\Property(
+     *                 property="error",
+     *                 type="object",
+     *                 @OA\Property(property="status", type="string", default="400"),
+     *                 @OA\Property(property="message", type="string", default="Share is already liked")
+     *             )
+     *         )
+     *     )
+     * )
+     * @inheritDoc
+     */
+    public function create(): EndpointResult
+    {
+        $shareId = $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            self::PARAMETER_SHARE_ID
+        );
+
+        $buzzShare = $this->getBuzzService()->getBuzzDao()->getBuzzShareById($shareId);
+        if (!$buzzShare instanceof BuzzShare) {
+            throw $this->getInvalidParamException(self::PARAMETER_SHARE_ID);
+        }
+
+        $buzzShareOnLike = $this->getBuzzService()
+            ->getBuzzLikeDao()
+            ->getBuzzLikeOnShareByShareIdAndEmpNumber($shareId, $this->getAuthUser()->getEmpNumber());
+        if ($buzzShareOnLike instanceof BuzzLikeOnShare) {
+            throw $this->getBadRequestException('Share is already liked');
+        }
+
+        $buzzShare->getDecorator()->increaseNumOfLikesByOne();
+        $this->getBuzzService()->getBuzzDao()->saveBuzzShare($buzzShare);
```

Comment:
should wrap `create` and `delete` in a transaction.
e.g. https://github.com/orangehrm/orangehrm/pull/1535/files#diff-059e542f32bb7cdac65aaa45dfad0a2bbed812979b77a58a333190cce849779aR156-R181

Reply:
ah yes forgot that 🤦 . will add it